src/app_engine/apprtc.py

```diff
@@ -546,6 +546,110 @@ def get(self):
     params = get_room_parameters(self.request, None, None, None)
     self.response.write(json.dumps(params))
 
+class AnalyticsPage(webapp2.RequestHandler):
+  """Client Analytics data handler.
+
+  Each POST body to the AnalyticsPage is a JSON object of the form,
+  {
+    'request_time_ms': <client time in milliseconds when the request was sent>,
+    'type': <type of the content below>,
+    'content': <content related to the above type>
+  }
+
+  Currently the AnalyticsPage handles the following types:
+  'event': A JSON event object of the form:
+          {
+            'event_type': <string corresponding to an attribute of EventType>,
+            'event_time_ms': <client time when the event occurred>,
+            'room_id': <id of the room corresponding to the event [optional]>,
+          }
+
+      The event is reported to Analytics.report_event with the
+      timestamp of the event modified to take into consideration the
+      clock differences between the client and the
+      server. 'request_time_ms' is compared to server time to provide
+      an offset, and the recorded event 'timestamp' is 'event_time_ms'
+      adjusted by the calculated offset. The original 'event_time_ms'
+      is reported as the 'client_timestamp'. This ignores the latency
+      of opening a connection and sending the body of the message to
+      the Analytics page. Travel time should be negligible.
+  """
+  def _write_response(self, result, message=None):
+    response = {'result': result}
+    if message is not None:
+      response['message'] = message
+
+    self.response.write(json.dumps(response));
+
+  def _write_invalid_request(self):
+    self.error(400)
```

Comment:
The other GAE APIs does not return an error code except for server internal error. Can you just return 200?